tests/topotests/bgp_community_change_update/test_bgp_community_change_update.py

```diff
@@ -0,0 +1,226 @@
+from lib.topolog import logger
+from mininet.topo import Topo
+
+from lib.common_config import step
+from time import sleep
+
+
+class TemplateTopo(Topo):
+    def build(self, *_args, **_opts):
+        tgen = get_topogen(self)
+
+        tgen.add_router("z1")
+        tgen.add_router("y1")
+        tgen.add_router("y2")
+        tgen.add_router("y3")
+        tgen.add_router("x1")
+        tgen.add_router("c1")
+
+        # 10.0.1.0/30
+        switch = tgen.add_switch("s1")
+        switch.add_link(tgen.gears["c1"])
+        switch.add_link(tgen.gears["x1"])
+
+        # 10.0.2.0/30
+        switch = tgen.add_switch("s2")
+        switch.add_link(tgen.gears["x1"])
+        switch.add_link(tgen.gears["y1"])
+
+        # 10.0.3.0/30
+        switch = tgen.add_switch("s3")
+        switch.add_link(tgen.gears["y1"])
+        switch.add_link(tgen.gears["y2"])
+
+        # 10.0.4.0/30
+        switch = tgen.add_switch("s4")
+        switch.add_link(tgen.gears["y1"])
+        switch.add_link(tgen.gears["y3"])
+
+        # 10.0.5.0/30
+        switch = tgen.add_switch("s5")
+        switch.add_link(tgen.gears["y2"])
+        switch.add_link(tgen.gears["y3"])
+
+        # 10.0.6.0/30
+        switch = tgen.add_switch("s6")
+        switch.add_link(tgen.gears["y2"])
+        switch.add_link(tgen.gears["z1"])
+
+        # 10.0.7.0/30
+        switch = tgen.add_switch("s7")
+        switch.add_link(tgen.gears["y3"])
+        switch.add_link(tgen.gears["z1"])
+
+
+def setup_module(mod):
+    tgen = Topogen(TemplateTopo, mod.__name__)
+    tgen.start_topology()
+
+    router_list = tgen.routers()
+
+    for i, (rname, router) in enumerate(router_list.items(), 1):
+        router.load_config(
+            TopoRouter.RD_ZEBRA, os.path.join(CWD, "{}/zebra.conf".format(rname))
+        )
+        router.load_config(
+            TopoRouter.RD_BGP, os.path.join(CWD, "{}/bgpd.conf".format(rname))
+        )
+
+    tgen.start_router()
+
+
+def teardown_module(mod):
+    tgen = get_topogen()
+    tgen.stop_topology()
+
+
+def test_bgp_community_update_path_change():
+    tgen = get_topogen()
+
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    def _bgp_converge_initial():
+        output = json.loads(
+            tgen.gears["c1"].vtysh_cmd("show ip bgp neighbor 10.0.1.2 json")
+        )
+        expected = {
+            "10.0.1.2": {
+                "bgpState": "Established",
+                "addressFamilyInfo": {"ipv4Unicast": {"acceptedPrefixCounter": 8}},
+            }
+        }
+        return topotest.json_cmp(output, expected)
+
+    step("Check if an initial topology is converged")
+    test_func = functools.partial(_bgp_converge_initial)
+    success, result = topotest.run_and_expect(test_func, None, count=60, wait=0.5)
+    assert result is None, "Failed to see bgp convergence in c1"
+
+    step("Disable link between y1 and y2")
+    tgen.gears["y1"].run("ifconfig y1-eth1 down")
```

Comment:
please use ``ip link set ...`` instead of deprecated ``ifconfig``. BSD support has been given up for mininet years ago.

Reply:
Fixed, thanks.